lib/Runtime/Library/JavascriptPromise.cpp

```diff
@@ -932,7 +932,17 @@ namespace Js
         JavascriptExceptionObject* exception = nullptr;
 
         {
-            Js::JavascriptExceptionOperators::AutoCatchHandlerExists autoCatchHandlerExists(scriptContext);
+            Var promiseVar = promiseCapability->GetPromise();
```

Comment:
This is a perverse case, but what about this scenario:

```js
var p = Promise.resolve(0).then(() => {
  p.catch(() => {}); // lazily added catch on the currently executing promise
  throw "Am I uncaught?";
});
```

Reply:
It also looks to me like this will blame the exception after the fact:

```js
var p = Promise.resolve(0).then(() => {
    throw "foo";
}).then(() => { console.log("not executed, but no catch handler") });
```


Running that snippet, won't it initially think that there is a catch handler since the second `then` installs both (but a stub for the error case), and then when the error is passed through by the second then, this code would register as an uncaught exception at that time?

Reply:
I'll add a test for this.  I think in this case when the throw happens, there will be handler, so we wouldn't want to raise for "uncaught exceptions.

Reply:
This is where a lot of the discussion in #4630 went :) The hope being that when you are trying to inspect uncaught execptions, you want to see as much of the state at the point where the exception was thrown (to work out what's going on), while here we're already in a completely different async stack.